File: algorithms/code/leetcode/lc007_reverse_integer/LC007ReverseInteger.Tests.cs
using Xunit;

namespace AlgorithmDesign.code.leetcode.lc007_reverse_integer
{
    public class SolutionTests
    {
        [Fact]
        public void Test2000000002()
        {
            Assert.Equal(2000000002, new Solution().Reverse(2000000002));
        }

        [Fact]
        public void TestMinus2147483648()
        {
            Assert.Equal(0, new Solution().Reverse(-2147483648));
        }

        [Fact]
        public void Test1000000003()
        {
            Assert.Equal(0, new Solution().Reverse(1000000003));
        }

        [Fact]
        public void Test1534236469()
        {
            Assert.Equal(0, new Solution().Reverse(1534236469));
        }

        [Fact]
        public void TestMinus321()
        {
            Assert.Equal(-123, new Solution().Reverse(-321));
        }
    }
}

File: algorithms/code/leetcode/lc007_reverse_integer/LC007ReverseIntegerTests.java
package leetcode.lc007_reverse_integer;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC007ReverseIntegerTests {
    @Test
    public void test2000000002() throws Exception {
        LC007ReverseInteger solution;
        solution = new LC007ReverseInteger();
        assertEquals(2000000002, solution.reverse(2000000002));
    }

    @Test
    public void testMinus2147483648() throws Exception {
        LC007ReverseInteger solution;
        solution = new LC007ReverseInteger();
        assertEquals(0, solution.reverse(-2147483648));
    }

    @Test
    public void test1000000003() throws Exception {
        LC007ReverseInteger solution;
        solution = new LC007ReverseInteger();
        assertEquals(0, solution.reverse(1000000003));
    }

    @Test
    public void test1534236469() throws Exception {
        LC007ReverseInteger solution;
        solution = new LC007ReverseInteger();
        assertEquals(0, solution.reverse(1534236469));
    }

    @Test
    public void testMinus321() throws Exception {
        LC007ReverseInteger solution;
        solution = new LC007ReverseInteger();
        assertEquals(-123, solution.reverse(-321));
    }
}
